evm/contracts/examples/ConversionRate.sol

```diff
@@ -51,49 +62,63 @@ contract ConversionRate is ChainlinkClient, Ownable {
    */
   function requestRateUpdate()
     public
+    ensureAuthorizedRequester()
   {
     Chainlink.Request memory request;
     bytes32 requestId;
+    uint256 oraclePayment = paymentAmount;
 
     for (uint i = 0; i < oracles.length; i++) {
       request = buildChainlinkRequest(jobIds[i], this, this.chainlinkCallback.selector);
-      requestId = sendChainlinkRequestTo(oracles[i], request, ORACLE_PAYMENT);
+      requestId = sendChainlinkRequestTo(oracles[i], request, oraclePayment);
       requestAnswers[requestId] = answerCounter;
     }
-    answers[answerCounter].expectedResponses = oracles.length;
+    answers[answerCounter].minimumResponses = minimumResponses;
+    answers[answerCounter].maxResponses = oracles.length;
     answerCounter = answerCounter.add(1);
   }
 
   /**
    * @notice Receives the answer from the Chainlink node.
    * @dev This function can only be called by the oracle that received the request.
    * @param _clRequestId The Chainlink request ID associated with the answer
-   * @param _rate The answer provided by the Chainlink node
+   * @param _response The answer provided by the Chainlink node
    */
-  function chainlinkCallback(bytes32 _clRequestId, uint256 _rate)
+  function chainlinkCallback(bytes32 _clRequestId, uint256 _response)
     public
   {
     validateChainlinkCallback(_clRequestId);
 
     uint256 answerId = requestAnswers[_clRequestId];
     delete requestAnswers[_clRequestId];
-    answers[answerId].responses.push(_rate);
 
-    updateRecords(answerId);
+    answers[answerId].responses.push(_response);
+    updateLatestAnswer(answerId);
+    deleteAnswer(answerId);
```

Comment:
Does this need to be in a separate method with the `ensureAllResponsesReceived` modifier if you're already in the callback? Do you deliberately want to NOT delete if you get a callback without all responses received or should that be an invariant (impossibility)?

Reply:
Correct, don't delete until the answer can be as accurate as possible.